### ISSUE TITLE
Missing Error Handling?

### ISSUE THREAD

**Issue by Swader** [Webiny/Tutorial-TodoApp#1]
Tried adding new task in Todo App, didn't work. Missing error reporting?

Inspected console, noticed `Uncaught TypeError: Cannot read property 'id' of undefined
(index):128 Uncaught TypeError: Cannot read property 'id' of undefined" when grabbing "data.data" from the`task` variable. Another case where error checking is necessary but missing.

![02](https://cloud.githubusercontent.com/assets/1430603/5692015/49ca6c54-98e3-11e4-8071-65aae66f74ae.PNG)

Looked into network call to API, seems like the entire framework's end is missing error reporting - the templating engine is throwing uncaught exceptions.

![03](https://cloud.githubusercontent.com/assets/1430603/5692016/60a37a10-98e3-11e4-8a9a-250e8b582dae.PNG)

In summary, there are three locations where error checking/reporting is missing - fixing just one will make UX better, but to get an optimal experience all three should be fixed.
